### PR TITLE
added sysl-safe names checker and converter

### LINKED ISSUE #499: importing query params with hyphens generate bad sysl
Attempting to import a swagger/openAPI spec where a query parameter contains a '-' causes the generated sysl to contain the - also, which is invalid.

example:
```
- name: cust-nums
     type: string
     required: true
     in: query
```

generated out to `?cust-nums=string&....` The python importer would convert that to `custNums=string` which is now valid sysl but no longer follows the spec.

### PR DESCRIPTION
**Pull request by nofun97**
Fixes #499 .

Changes proposed in this pull request:
- Added a functionality to check unsafe parameter names and convert it

currently it checks for hyphens and spaces only.

@anz-bank/sysl-developers

### CODE REVIEW

File: pkg/importer/openapi.go [hunk 1]
```diff
@@ -337,10 +337,14 @@ func (l *loader) buildParams(params openapi3.Parameters) Parameters {
 }
 
 func (l *loader) buildParam(p *openapi3.Parameter) Param {
+	name := p.Name
+	if hasToBeGoSafe(p.In) {
```
Comment (anz-gordonj7) [inline pkg/importer/openapi.go]: this actually needs to be sysl safe, not go safe. the - breaks the sysl parser

File: pkg/importer/openapi.go [hunk 2]
```diff
@@ -349,3 +353,32 @@ func (l *loader) buildParam(p *openapi3.Parameter) Param {
 		In: p.In,
 	}
 }
+
+func hasToBeGoSafe(in string) bool {
+	return strings.ToLower(in) == "query"
+}
+
+func convertToGoSafe(name string) string {
```
Comment (anz-gordonj7) [inline pkg/importer/openapi.go]: this func is a bit gross to be honest. You could probably make it much simpler by doing `strings.Split(name, "-")` and then rejoining with some smarts.

File: pkg/importer/tests-swagger/EXAMPLE_SWAGGER_UNSAFE_PARAM_NAME.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
+swagger: "2.0"
```
Comment (anz-gordonj7) [inline pkg/importer/tests-swagger/EXAMPLE_SWAGGER_UNSAFE_PARAM_NAME.yaml]: EXCELLENT! very thorough testing

File: pkg/importer/openapi.go (hunk 2 above)
Comment (nofun97) [reply pkg/importer/openapi.go]: i thought of that, but i think with the current function we can support more unsafe characters later if needed, what do you think?

File: pkg/importer/openapi.go (hunk 1 above)
Comment (nofun97) [reply pkg/importer/openapi.go]: changed to sysl safe

File: pkg/importer/openapi.go (hunk 2 above)
Comment (anz-gordonj7) [reply pkg/importer/openapi.go]: its fine

**Pull request merged**